tests/b2share_functional_tests/test_record_cli.py

```diff
@@ -0,0 +1,94 @@
+
+import json
+from click.testing import CliRunner
+from flask_cli import ScriptInfo
+from flask import url_for
+
+from b2share.modules.communities.api import Community
+from b2share.modules.deposit.api import PublicationStates
+from b2share.modules.records.cli import check_dois
+from b2share_unit_tests.helpers import create_user, generate_record_data
+
+headers = [('Content-Type', 'application/json'),
+           ('Accept', 'application/json')]
+patch_headers = [('Content-Type', 'application/json-patch+json'),
+                 ('Accept', 'application/json')]
+
+def test_record_doi_cli(app, test_communities, login_user):
+    """Test adding a community and setting its schema using CLI commands."""
+    with app.app_context():
+        community_name = test_communities.popitem()[0]
+        community = Community.get(name=community_name)
+        com_admin = create_user('com_admin', roles=[community.admin_role])
+        test_records_data = [generate_record_data(community=community_name)
+                             for _idx in range(1,3)]
+
+        record_id_list = []
+        for record_data in test_records_data:
+            with app.test_client() as client:
+                login_user(com_admin, client)
+
+                draft_create_res = client.post(url_for('b2share_records_rest.b2rec_list'),
+                                               data=json.dumps(record_data),
+                                               headers=headers)
+                assert draft_create_res.status_code == 201
+                draft_create_data = json.loads(
+                    draft_create_res.get_data(as_text=True))
+                record_id = draft_create_data['id']
+                draft_url = url_for('b2share_deposit_rest.b2dep_item',
+                                    pid_value=record_id)
+                draft_submit_res = client.patch(
+                    draft_url,
+                    headers=patch_headers,
+                    data=json.dumps([{
+                        "op": "replace", "path": "/publication_state",
+                        "value": PublicationStates.submitted.name
+                    }]))
+                assert draft_submit_res.status_code == 200
+                draft_publish_res = client.patch(
+                    draft_url,
+                    headers=patch_headers,
+                    data=json.dumps([{
+                        "op": "replace", "path": "/publication_state",
+                        "value": PublicationStates.published.name
+                    }]))
+                assert draft_publish_res.status_code == 200
+                record_id_list.append(record_id)
+
+
+        runner = CliRunner()
+        script_info = ScriptInfo(create_app=lambda info: app)
+
+        # Run 'check_record_doi' command for all records
+        result = runner.invoke(check_dois, ['-a'], obj=script_info)
+        if result.exit_code != 0:
+            print(result.output)
+        assert result.exit_code == 0
```

Comment:
This does not test if the command succeeds, just that it does not return an error code.